starter-code/js/stack.js

```diff
@@ -0,0 +1,204 @@
+function StackObject(){
+  this.structureLength = 10;
+  this.structureArray = [];
+  this.numObjs = 0;
+}
+
+StackObject.prototype._overFlow = function(){
+
+  if(this.numObjs>=this.structureLength)
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+
+};
+
+
+StackObject.prototype._underFlow = function(){
+
+  if(this.numObjs===0)
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+
+};
+
+StackObject.prototype.pushObj = function(element){
+
+  if(this._overFlow())
+  {
+    return ["null",false];
+  }
+  else
+  {
+    var elementObj = {obj: element , index: this.numObjs };
+    this.numObjs++;
+    this.structureArray.push(elementObj);
+
+    return [elementObj,true];
+  }
+
+};
+
+StackObject.prototype.popObj = function(){
+
+  if(this._underFlow())
+  {
+    return ["null",false];
+  }
+  else
+  {
+    var tempArray = [];
+    var temp = "";
+    var self = this;
+
+    this.structureArray.forEach(function(elementObj){
+
+      if(elementObj.index !== self.numObjs-1)
+      {
+        tempArray.push(elementObj);
+      }
+      else {
+
+        temp = elementObj;
+
+      }
+    });
+
+    this.structureArray = tempArray;
+    this.numObjs--;
+
+    return [temp,true];
+  }
+
+};
+
+
+//////////////////////////jquery///////////////////////////////////////////
+
+var stackObjectDoc;
+var addObjectDocStack="";
+
+$(document).ready(function(){
+
+  stackObjectDoc = new StackObject();
+  generateBoardHtml(stackObjectDoc,"stack");
+  stackBoard(addObjectDocStack, stackObjectDoc);
+
+
+});
+
```

Comment:
instead of writing the html directly you should try to create the html elements through jQuery -> $('<div>').addClass('someClass')